### PR TITLE
Fix build on NetBSD

### PR DESCRIPTION
**Pull request by niacat**

### CODE REVIEW

File: src/file.cc [hunk 1]
```diff
@@ -21,7 +21,7 @@
 #include <sys/select.h>
 #include <unistd.h>
 
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD__) or defined(__NetBSD__)
```
Comment (lenormf) [inline src/file.cc]: Does the pre-processor support that?

File: src/file.cc [hunk 2]
```diff
@@ -624,7 +624,7 @@ String get_kak_binary_path()
     kak_assert(res != -1);
     buffer[res] = '\0';
     return buffer;
-#elif defined(__FreeBSD__)
+#elif defined(__FreeBSD__) or defined(__NetBSD__)
     int mib[] = {CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1};
```
Comment (krytarowski) [inline src/file.cc:628]: NetBSD muse use different arguments to mib[]

**Comment by krytarowski** [comment]
Thanks. Now it would be nice to fix the `mib[]` arguments and pass `CTL_KERN, KERN_PROC_ARGS, -1, KERN_PROC_PATHNAME` for NetBSD.

**Comment by mawww** [comment]
Do you know if that would be compatible with FreeBSD or if we need to split the logic between NetBSD and FreeBSD ?

**Comment by krytarowski** [comment]
We need to ifdef each OS separately.

**Comment by mawww** [comment]
I do not have NetBSD available for testing, could you create a pull request with such a change ?

**Comment by krytarowski** [comment]
https://github.com/mawww/kakoune/pull/3453

**Pull request merged**